cpp/oneapi/dal/table/backend/csr_kernels.cpp

```diff
@@ -322,6 +325,188 @@ void csr_pull_block(const Policy& policy,
     }
 }
 
+#ifdef ONEDAL_DATA_PARALLEL
+
+std::int64_t csr_get_non_zero_count(sycl::queue& queue,
+                                    const std::int64_t row_count,
+                                    const std::int64_t* row_offsets,
+                                    const std::vector<sycl::event>& dependencies) {
+    if (row_count <= 0)
+        return 0;
+
+    std::int64_t first_row_offset{ 0L }, last_row_offset{ 0L };
+    auto first_row_event = copy_usm2host(queue, &first_row_offset, row_offsets, 1, dependencies);
+    auto last_row_event =
+        copy_usm2host(queue, &last_row_offset, row_offsets + row_count, 1, dependencies);
+    sycl::event::wait_and_throw({ first_row_event, last_row_event });
+
+    return (last_row_offset - first_row_offset);
+}
+
+#endif
+
+std::int64_t csr_get_non_zero_count(const std::int64_t row_count, const std::int64_t* row_offsets) {
+    if (row_count <= 0)
+        return 0;
```

Comment:
Do you really want to support tables with zero rows?

Reply:
It's more like out-of-bounds check.
CSR tables with 0 rows or columns are not supported and the respective error handling is present in the code.